src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -107,6 +107,9 @@ class RscCompile(ZincCompile, MirroredTargetOptionMixin):
   _name = 'mixed' # noqa
   compiler_name = 'rsc'
 
+  deprecated_options_scope = 'compile.zinc'
```

Comment:
Related to this: there is one more thing that we need to do.

As you've probably noticed: declaring a deprecated options scope like this won't also cause `rsc` to end up consuming the `cache-compile-zinc` options, because this task won't declare a dependency on that scope. It's very good that you went and fixed all the usages, because even if that substem actually existed, we'd want to fix the references.

But I think that we should ensure that that subsystem actually exists so that folks get a deprecation warning.

To do that, you can use "deprecated subsystem dependency":
```
def subsystem_dependencies(cls):
  ...
  dep = CacheSetup.scoped(
      Zinc,
      removal_version='1.19.0.dev0',
      removal_hint='Cache options for `zinc` should move to `rsc`.'
    )
```
... this will declare the subsystem, and will trigger a deprecation when it is used, but it _won't_ actually cause you to use those options. I think that this is the best that we can do without a bunch of hacks, so I think it's worth going for.

Reply:
(this is the only blocker for landing this review, IIRC.)

Reply:
implemented below: